### ISSUE TITLE
v2.7.4 has dirty git tree state in version output

### ISSUE THREAD

**Issue by sudermanjr** [argoproj/argo#2725]
Checklist:

 * [x] I've included the version.
 * [x] I've included reproduction steps.
 * [ ] I've included the workflow YAML.
 * [ ] I've included the logs.
 
**What happened**:

```
└─ argo version
argo: vv2.7.4+50b209c.dirty
  BuildDate: 2020-04-16T16:28:14Z
  GitCommit: 50b209ca14c056fb470ebb8329e255304dd5be90
  GitTreeState: dirty
  GitTag: v2.7.4
  GoVersion: go1.13
  Compiler: gc
  Platform: darwin/amd64
```

**What you expected to happen**:

I would expect to see something more like the output from 2.7.2

```
argo: v2.7.2
  BuildDate: 2020-04-10T19:25:22Z
  GitCommit: c52a65aa62426f5e874e1d3f1058af15c43eb35f
  GitTreeState: clean
  GitTag: v2.7.2
  GoVersion: go1.13.4
  Compiler: gc
  Platform: darwin/amd64
```

**How to reproduce it (as minimally and precisely as possible)**:
`argo version`

---

**Message from the maintainers**:

If you are impacted by this bug please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by alexec** [comment]
I've noticed that both dirty and clean builds seem to have this problem.

**Comment by alexec** [comment]
Should be fixed in v2.7.5.
